### PR TITLE
Loki ringstatus

### PR DESCRIPTION
**Pull request by hervenicol**
This PR:

- adds a loki alert about ringStatus. It should trigger earlier than the others and in a more precise way. And help with incidents analysis. This alert won't trigger at night though. At least for the moment.
- adds unit tests for Loki alerts

Towards https://github.com/giantswarm/giantswarm/issues/23284 - should help with diagnostics

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).
- [ ] Alerting rules must have a comment documenting why it needs to exist.
- [ ] Consider creating a dashboard (if it does not exist already) to help oncallers monitor the status of the issue.

### CODE REVIEW

**Review by QuentinBisson — APPROVED** [review]

File: test/tests/providers/global/loki.all.rules.test.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        values: "0+0x20 1+0x160"  # 1 unhealthy value after 20 minutes
+      - series: 'loki_panic_total{app="loki-compactor", cluster_id="zj88t", cluster_type="workload_cluster", container="compactor", customer="giantswarm", installation="gorilla", instance="10.7.116.221:3100", job="zj88t-prometheus/workload-zj88t/0", namespace="loki", node="ip-10-6-2-178.eu-central-1.compute.internal", organization="giantswarm-production", pod="loki-compactor-676b8c897b-rq298", provider="aws", service_priority="highest"}'
+        values: "0+0x20 1+0x160"  # 1 panic after 20 minutes
+      - series: 'loki_request_duration_seconds_count{app="loki-distributor", cluster_id="zj88t", cluster_type="workload_cluster", container="distributor", customer="giantswarm", installation="gorilla", instance="10.7.75.90:3100", job="zj88t-prometheus/workload-zj88t/0", method="POST", namespace="loki", node="ip-10-6-2-141.eu-central-1.compute.internal", organization="giantswarm-production", pod="loki-distributor-74b78f5559-tz6zs", provider="aws", route="loki_api_v1_push", service_priority="highest", status_code="204", ws="false"}'
+        values: "0+60x180"  # 1 request per second OK for 3 hours
```
Comment (TheoBrigitte) [inline test/tests/providers/global/loki.all.rules.test.yml:13]: IIUC `0+60x180` means we add 60 to each sample, giving us something like

```
0 60 120 180 240 ...
```

I read this as we have 60 request per second for 3 hours, am I wrong ?

File: test/tests/providers/global/loki.all.rules.test.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      - series: 'loki_request_duration_seconds_count{app="loki-distributor", cluster_id="zj88t", cluster_type="workload_cluster", container="distributor", customer="giantswarm", installation="gorilla", instance="10.7.75.90:3100", job="zj88t-prometheus/workload-zj88t/0", method="POST", namespace="loki", node="ip-10-6-2-141.eu-central-1.compute.internal", organization="giantswarm-production", pod="loki-distributor-74b78f5559-tz6zs", provider="aws", route="loki_api_v1_push", service_priority="highest", status_code="503", ws="false"}'
+        values: "0+0x20 0+30x160"  # After 20 minutes, we also have 0.5 rq/s failing 
+    alert_rule_test:
+
+      - alertname: LokiRequestPanics
+        eval_time: 15m  # should be OK after 15 minutes
+        exp_alerts:
```
Comment (TheoBrigitte) [inline test/tests/providers/global/loki.all.rules.test.yml:20]: an empty `exp_alerts` mean the alert does not fire right ?

File: test/tests/providers/global/loki.all.rules.test.yml (hunk 1 above)
Comment (hervenicol) [reply test/tests/providers/global/loki.all.rules.test.yml:13]: We defined `interval: 1m`, so it's one sample per minute. So 60 requests per minute.

File: test/tests/providers/global/loki.all.rules.test.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        values: "0+60x180"  # 1 request per second OK for 3 hours
+      - series: 'loki_request_duration_seconds_count{app="loki-distributor", cluster_id="zj88t", cluster_type="workload_cluster", container="distributor", customer="giantswarm", installation="gorilla", instance="10.7.75.90:3100", job="zj88t-prometheus/workload-zj88t/0", method="POST", namespace="loki", node="ip-10-6-2-141.eu-central-1.compute.internal", organization="giantswarm-production", pod="loki-distributor-74b78f5559-tz6zs", provider="aws", route="loki_api_v1_push", service_priority="highest", status_code="503", ws="false"}'
+        values: "0+0x20 0+30x160"  # After 20 minutes, we also have 0.5 rq/s failing
```
[the remark this answers is not in the document]
Comment (hervenicol) [reply test/tests/providers/global/loki.all.rules.test.yml:15]: so here, 30 rq/min ;)

File: test/tests/providers/global/loki.all.rules.test.yml (hunk 2 above)
Comment (hervenicol) [reply test/tests/providers/global/loki.all.rules.test.yml:20]: Right!

**Pull request merged**